### PR TITLE
Updated recipe for version sentence-transformer version 2.1.0

### PR DESCRIPTION
**Pull request by sugatoray**
The following are the list of activities covered in this PR. The absence of `torchvision` library in the feedstock recipe was causing problem with [another PR](https://github.com/conda-forge/staged-recipes/pull/16963). Please merge this PR (once it passes all the CI's).

This closes Issue #22. 

- [x] updated sha256
- [x] added dependencies:
	- [x] `torchvision`
	- [x] `tokenizers`
	- [x] `sentencepiece`
	- [x] `huggingface_hub`
- [x] updated/corrected versions of:
	- [x] `pytorch`
	- [x] `transformers`
- [x] updated test (pip check)
- [x] added @sugatoray as a maintainer

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged) `NA` (changed version: `2.1.0`)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by sugatoray** [comment]
@conda-forge-admin, please rerender

**Comment by sugatoray** [comment]
cc: @ndmaxar, @oblute, @rluria14

**Comment by sugatoray** [comment]
gentle reminder: @ndmaxar, @oblute, @rluria14

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,14 +19,22 @@ requirements:
...
+    - pytorch >=1.6.0
+    - torchvision
+    - sentencepiece
+    - tokenizers >=0.10.3
+    - transformers >=4.6.0,<5.0.0
+  run_constrained:
+    # NOTE: huggingface-hub is not yet available on conda-forge.
+    # Moving huggingface-hub under "run_constrained" as a temporary 
+    # fix, until the package is made available on conda-forge.
+    - huggingface-hub
```
Comment (BastianZim) [inline recipe/meta.yaml]:
```suggestion
huggingface_hub
```

The package is on conda with `_` instead of `-`.

**Review by BastianZim — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (sugatoray) [reply recipe/meta.yaml]: Good catch! Thanks.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,14 +19,18 @@ requirements:
...
     - scipy
-    - pytorch >=1.0.1
     - tqdm
-    - transformers >=3.1
+    - pytorch >=1.6.0
+    - torchvision
+    - sentencepiece
+    - tokenizers >=0.10.3
+    - transformers >=4.6.0,<5.0.0
+huggingface_hub
```
Comment (BastianZim) [inline recipe/meta.yaml]:
```suggestion
    - huggingface_hub
```
That was my bad, sorry.

Comment (sugatoray) [reply recipe/meta.yaml]: No problem. I took care of it. :)

**Comment by sugatoray** [comment]
@conda-forge-admin, please ping conda-forge/core

**Review by xhochy — APPROVED** [review]

**Pull request merged**